client/extensions/woocommerce/state/sites/products/actions.js

```diff
@@ -12,10 +12,10 @@ import {
  * @param {Number} siteId The id of the site upon which to create the product.
  * @param {Object} product The complete product object (may include a placeholder id)
  * @param {String} [successAction=undefined] Optional action object to be dispatched upon success.
- * @param {String} [errorAction=undefined] Optional action object to be dispatched upon error.
+ * @param {String} [failureAction=undefined] Optional action object to be dispatched upon error.
  * @return {Object} Action object
  */
-export function createProduct( siteId, product, successAction = undefined, errorAction = undefined ) {
```

Comment:
There's no need for the `=undefined`, it's implicit.

Also, there's no need for the `if ( failureAction ) { action.failureAction = failureAction }`. Just set `action.failureAction` always. If `failureAction` is `undefined`, the resulting object will be the same (for our purposes).